test/common/http/path_utility_test.cc

```diff
@@ -85,5 +85,24 @@ TEST_F(PathUtilityTest, NormalizeCasePath) {
 // "/../c\r\n\"  '\n' '\r' should be excluded by http parser
 // "/a/\0c",     '\0' should be excluded by http parser
 
+// Paths that are valid get normalized.
+TEST_F(PathUtilityTest, MergeSlashes) {
+  const std::vector<std::pair<std::string, std::string>> non_normal_pairs{
+      {"", ""},                        // empty
+      {"/a", "/a"},                    // no-op
+      {"//a/b/c", "/a/b/c"},           // double / start
+      {"/a//b/c", "/a/b/c"},           // double / in the middle
+      {"/a//b?a=///c", "/a/b?a=///c"}, // slashes in the query are ignored
+      {"/a//b?", "/a/b?"},             // empty query
+  };
+
+  for (const auto& path_pair : non_normal_pairs) {
+    auto& path_header = pathHeaderEntry(path_pair.first);
+    PathUtil::mergeSlashes(path_header);
+    EXPECT_EQ(path_header.value().getStringView(), path_pair.second)
+        << "original path: " << path_pair.first;
+  }
+}
```

Comment:
include a test method that also ensures that no slash-merging takes place when the option is off.

Reply:
It's covered by `ConnectionManagerUtilityTest::MergeSlashesDefaultOff` test

Reply:
I see, ok, that makes sense. I think handling of 3 or more slashes should be covered here, though, as this is the functional unit-test.

Reply:
Test for handling of 3 or more slashes was already added in the latest revision